pallets/runtime/tests/src/transaction_ordering_test.rs

```diff
@@ -0,0 +1,1056 @@
+            let (instruction_counter1000, alice_sent_amount_1000, _bob_received_amount_1000) =
+                initialize_transaction(
+                    alice_secret_account.clone(),
+                    alice_creds.clone(),
+                    alice_init_balance.clone(),
+                    bob_secret_account.clone(),
+                    bob_creds.clone(),
+                    charlie_creds.clone(),
+                    tx_id.clone(),
+                    10,
+                );
+            let alice_pending_balance = alice_init_balance - alice_sent_amount_1000;
+
+            let tx_id = tx_id + 1;
+            let (instruction_counter1001, dave_sent_amount_1001, alice_received_amount_1001) =
+                initialize_transaction(
+                    dave_secret_account.clone(),
+                    dave_creds.clone(),
+                    dave_init_balance.clone(),
+                    alice_secret_account.clone(),
+                    alice_creds.clone(),
+                    charlie_creds.clone(),
+                    tx_id.clone(),
+                    8,
+                );
+
+            let tx_id = tx_id + 1;
+            let (instruction_counter1002, alice_sent_amount_1002, dave_received_amount_1002) =
+                initialize_transaction(
+                    alice_secret_account.clone(),
+                    alice_creds.clone(),
+                    alice_pending_balance.clone(),
+                    dave_secret_account.clone(),
+                    dave_creds.clone(),
+                    charlie_creds.clone(),
+                    tx_id.clone(),
+                    14,
+                );
+
+            // Approve and process tx:1001.
+            finalize_transaction(
+                instruction_counter1001,
+                dave_creds.clone(),
+                alice_creds.clone(),
+                charlie_creds.clone(),
+                charlie_secret_account.clone(),
+                dave_init_balance - dave_sent_amount_1001,
+                alice_init_balance + alice_received_amount_1001,
+                false,
+            );
+
+            // Alice has a change of heart and rejects the transaction to Bob!
+            assert_ok!(Settlement::unauthorize_instruction(
+                Origin::signed(alice_creds.key.public()),
+                instruction_counter1000,
+                default_portfolio_vec(alice_creds.did)
+            ));
+
+            // Approve and process tx:1002.
+            finalize_transaction(
+                instruction_counter1002,
+                alice_creds.clone(),
+                dave_creds.clone(),
+                charlie_creds.clone(),
+                charlie_secret_account.clone(),
+                alice_init_balance + alice_received_amount_1001 - alice_sent_amount_1002,
+                dave_init_balance - dave_sent_amount_1001 + dave_received_amount_1002,
+                false,
+            );
+        });
+}
+
+#[test]
```

Comment:
Another scenario to test (could be a different story): what if a receiver never responds? What happens to the pending states? Do we have an expiry policy for such cases?

Reply:
Good question. If that happens the sender can call `settlement::unauthorize_instruction()` (which I'm also testing in `mercat_whitepaper_scenario1()`); it will take the transaction out of the pending state. So as is right now there's a workaround for it.

Reply:
Awesome!